sklearn/decomposition/pca.py

```diff
@@ -330,7 +330,7 @@ def get_covariance(self):
         components_ = self.components_
         exp_var = self.explained_variance_
         if self.whiten:
-            components_ = components_ * np.sqrt(exp_var[:, np.newaxis])
+            components_ *= np.sqrt(exp_var[:, np.newaxis])
```

Comment:
I am quite convinced that this line is wrong: it modifies 'components_', and this is a reference on 'self.components_'. As a result,self.components_ is now multiplied by the standard deviation.

Thus the logic of the code has been changed.